compiler/rustc_middle/src/mir/mono.rs

```diff
@@ -54,6 +54,33 @@ pub enum MonoItem<'tcx> {
     GlobalAsm(ItemId),
 }
 
+fn opt_incr_drop_glue_mode<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>) -> InstantiationMode {
+    // Non-ADTs can't have a Drop impl. This case is mostly hit by closures whose captures require
+    // dropping.
+    let Some(adt_def) = ty.ty_adt_def() else {
```

Comment:
nit: I prefer `let ty::Adt(adt_def, _) = ty.kind()` over `ty_adt_def`.

Reply:
I agree that's better.